File: vue/apps/web-antd/src/api/system/oper-log.ts
/**
 * @description 操作日志相关接口
 * @Link  https://github.com/xiujiecn/xiu-admin
 * @Copyright  Copyright (c) 2025 LiXiujie
 * @Author  Lxj <li@xiujie.cn>
 * @License  https://github.com/xiujiecn/xiu-admin/blob/master/LICENSE
 * @date 2024-03-21
 */

import { requestClient } from '#/api/request';

export interface SysOperLogListReq {
    operIp: string;
  businessType: string;
  method: string;
}

export interface SysOperLog {
    operId: number;
    tenantId: string;
    title: string;
    businessType: number;
    method: string;
    requestMethod: string;
    operatorType: number;
    operName: string;
    deptName: string;
    operUrl: string;
    operIp: string;
    operLocation: string;
    operParam: string;
    jsonResult: string;
    status: number;
    errorMsg: string;
    operTime: string;
    costTime: number;
}

export interface SysOperLogListRes {
  items: SysOperLog[];
  total: number;
}

export async function getSysOperLogList(params: SysOperLogListReq) {
    return requestClient.get<SysOperLogListRes>('/system/oper-log/list', { params });
}

File: server/api/system/v1/oper_log.go
// Package v1
// @Link  https://github.com/xiujiecn/xiu-admin
// @Copyright  Copyright (c) 2025 LiXiujie
// @Author  Lxj <li@xiujie.cn>
// @License  https://github.com/xiujiecn/xiu-admin/blob/master/LICENSE
package v1

import (
	"xiuadmin/internal/model"
	"xiuadmin/internal/model/request"
	"xiuadmin/internal/model/response"

	"github.com/gogf/gf/v2/frame/g"
)

type GetOperLogListReq struct {
	g.Meta `path:"/oper-log/list" method:"get" tags:"系统-日志管理" summary:"操作日志列表" x-check-permission:"cpm:monitor:operlog:list"`
	model.SysOperLogListParam
	request.PageInfo
}

type GetOperLogListRes struct {
	Items []*model.SysOperLogListModel `json:"items"`
	response.PageResult
}
